server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -1064,7 +1063,7 @@ public static StreamInput wrap(byte[] bytes, int offset, int length) {
      * Reads a vint via {@link #readVInt()} and applies basic checks to ensure the read array size is sane.
      * This method uses {@link #ensureCanReadBytes(int)} to ensure this stream has enough bytes to read for the read array size.
      */
-    private int readArraySize() throws IOException {
+    protected int readArraySize() throws IOException {
```

Comment:
is this necessary?

Reply:
Nah it was just for the string optimization -> reverted :)